### PR TITLE
snowflake: init at 1.1.0

### PR DESCRIPTION
**Pull request by witchof0x20**
###### Motivation for this change
!!!WIP!!!
Add snowflake, a Tor pluggable transport. The derivation is mostly identical to the one for obfs4.

I asked the upstream developers about packaging considerations and a couple things need to be done here before it's ready for push. Posting the mostly done WIP to ask for feedback / assistance.

- One file in this project (common/nat/nat.go) has a separate license from the the rest. Rather than being BSD3, it's licensed under the expat license. I'm not sure how to indicate this in the derivation.
- The developers also mentioned that only 2 of the generated executables from building the Go module should be included in the package. I'm not sure how to do this using `buildGoModule`
- A module would be nice to allow users to run snowflake servers more easily, but I think that is more suitable for another PR?

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -11996,6 +11996,16 @@
     githubId = 42300264;
     name = "wishfort36";
   };
+  witchof0x20 = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:11999]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/tools/networking/snowflake/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  vendorSha256 = "15nzqibrymbbn6cwz3267jxk60xr5f6v3akwplhjzcc16bgrcx57";
+
+  meta = with lib; {
+    description = "A pluggable transport proxy";
+    homepage = "https://snowflake.torproject.org";
+    maintainers = with maintainers; [ witchof0x20 ];
+    license = with lib.licenses; [
+      # common/nat/nat.go is licensed under the Expat license
+      mit
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/snowflake/default.nix:21]: expat is not mit, right?

File: pkgs/tools/networking/snowflake/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    sha256 = "0d5ddhg2p0mbcj1cmklwn04za2x1khxgm5x9qlsg1ywkn6ngnxad";
+  };
+
+  vendorSha256 = "15nzqibrymbbn6cwz3267jxk60xr5f6v3akwplhjzcc16bgrcx57";
+
+  meta = with lib; {
+    description = "A pluggable transport proxy";
+    homepage = "https://snowflake.torproject.org";
+    maintainers = with maintainers; [ witchof0x20 ];
+    license = with lib.licenses; [
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/snowflake/default.nix:19]:
```suggestion
    license = with licenses; [
```
already done by the with.

**Comment by ju1m** [comment]
@witchof0x20, thanks for this PR! I'm closing it though because in the meantime `snowflake` was packaged in https://github.com/NixOS/nixpkgs/pull/155457 .
